### ISSUE TITLE
deb-get prettylist is broken in main but fine in last release (3.6)

### ISSUE THREAD

**Issue by kingeejit** [wimpysworld/deb-get#551]
## Reporting a bug 🐛

This refers to deb-get currently in the main branch of the repo: 

`commit 180cbc1e737488b79bc9f296d10e2eef765499c8 (HEAD -> main, origin/main, origin/HEAD)`

The latest release (0.3.6) is unaffected.

The pretty list function exits with an error if the application is hosted on github and has a cache json file stored in /var/cache/deb-get/

This affects the ability to regenerate README.md with cog.

### Expected behavior

`deb-get prettylist` should complete successfully, printing out a markdown list of applications supported.

### Actual behavior

The pretty listing will stop with the first application that has been installed from github with and error message:

```
[!] ERROR! Missing required information of github package <package name>:
URL=
VERSION_PUBLISHED=
```
where package name is the apt application name used in `deb_<package name>` function.

### Steps to reproduce

1. Checkout deb-get (commit `180cbc1e737488b79bc9f296d10e2eef765499c8`)
2. Install a github project e.g. appimagelauncher since it's early in the list but not the first.
```
sudo ./deb-get install appimagelauncher
```
3. Generate prettylist using the github version of deb-get:
```
./deb-get prettylist
```

Output as follows:

```
| Source   | Package Name   | Description   |
| :------: | :------------- | :------------ |
| [<img src=".github/debian.png" align="top" width="20" />](https://1password.com/) | `1password` | <i>The easiest way to store and use strong passwords.</i> |
| [<img src=".github/github.png" align="top" width="20" />](https://github.com/OrangeDrangon/android-messages-desktop/) | `android-messages-desktop` | <i>Run Android Messages as a desktop app.</i> |
| [<img src=".github/github.png" align="top" width="20" />](https://antimicrox.github.io/) | `antimicrox` | <i>A graphical program used to map keyboard buttons and mouse controls to a gamepad.</i> |
| [<img src=".github/debian.png" align="top" width="20" />](https://anydesk.com/) | `anydesk` | <i>Access any device at any time. From anywhere. Always secure and fast.</i> |
  [!] ERROR! Missing required information of github package appimagelauncher:
URL=
VERSION_PUBLISHED=
```
#### /etc/os-release

```
PRETTY_NAME="Ubuntu 22.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.1 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

#### /etc/lsb-release

```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"
```

#### deb-get version

```
commit 180cbc1e737488b79bc9f296d10e2eef765499c
```

### deb-get update

Blushes, mutters "You didn't even buy me dinner!", opens kimono...

```
Hit:1 http://gb.archive.ubuntu.com/ubuntu jammy InRelease
Get:2 http://download.opensuse.org/repositories/home:/IBBoard:/cawbird/xUbuntu_22.04  InRelease [1,541 B]
Hit:3 http://gb.archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:4 http://gb.archive.ubuntu.com/ubuntu jammy-backports InRelease
Hit:5 http://security.ubuntu.com/ubuntu jammy-security InRelease
Hit:6 https://dl.google.com/linux/chrome/deb stable InRelease
Ign:7 https://repo.vivaldi.com/stable/deb stable InRelease
Hit:8 https://packages.microsoft.com/repos/edge stable InRelease
Hit:9 https://repo.vivaldi.com/stable/deb stable Release
Hit:10 https://packages.element.io/debian default InRelease
Get:11 https://packages.microsoft.com/repos/vscode stable InRelease [3,959 B]
Hit:12 https://ppa.launchpadcontent.net/apandada1/foliate/ubuntu jammy InRelease
Hit:13 https://ppa.launchpadcontent.net/cozy-team/cozy/ubuntu jammy InRelease
Hit:14 https://ppa.launchpadcontent.net/libreoffice/ppa/ubuntu jammy InRelease
Hit:15 https://ppa.launchpadcontent.net/papirus/papirus/ubuntu jammy InRelease
Hit:16 https://ppa.launchpadcontent.net/saiarcot895/chromium-beta/ubuntu jammy InRelease
Hit:17 https://ppa.launchpadcontent.net/yubico/stable/ubuntu jammy InRelease
Get:19 https://packages.microsoft.com/repos/vscode stable/main amd64 Packages [326 kB]
Fetched 332 kB in 8s (40.9 kB/s)
Reading package lists...
  [+] Updating /var/cache/deb-get/appimagelauncher.json
  [+] Updating /var/cache/deb-get/bitwarden.json
  [+] Updating /var/cache/deb-get/deb-get.json
  [+] Updating /var/cache/deb-get/flameshot.json
  [+] Updating /var/cache/deb-get/gh.json
  [+] Updating /var/cache/deb-get/github-desktop.json
  [+] Updating /var/cache/deb-get/heroic.json
  [+] Updating /var/cache/deb-get/openaudible.json
  [+] Updating /var/cache/deb-get/standard-notes.json
```

**Comment by kingeejit** [comment]
The issue appears to fall in the validate_deb function, specifically the logic on line 360:
```
        if (([ "${METHOD}" = github ] && [ -e "${CACHE_DIR}/${APP}.json" ]) || ([ "${METHOD}" != github ] && [ "${ACTION}" != "prettylist" ])) && ([ -z "${URL}" ] || [ -z "${VERSION_PUBLISHED}" ]); then
            fancy_message error "Missing required information of ${METHOD} package ${APP}:"
            echo "URL=${URL}"
            echo "VERSION_PUBLISHED=${VERSION_PUBLISHED}"
            exit 1
        fi
```

**Comment by natanjunges** [comment]
Already fixed by #544.
